src/backend/mpi/mpi.cpp

```diff
@@ -26,7 +28,13 @@ extern MPI_Comm workcomm;
  * @see swdsm.h
  * @see swdsm.cpp
  */
-extern MPI_Win  *globalDataWindow;
+extern std::vector<std::vector<MPI_Win>>  data_windows;
+/**
```

Comment:
Why are these `extern` declarations here instead of some appropriate header file (which is included here)?